drivers/volume/portworx/portworx.go

```diff
@@ -2757,6 +2769,35 @@ func (d *portworx) ToggleCallHome(n node.Node, enabled bool) error {
 	return nil
 }
 
+func (d *portworx) ValidateStorageCluster(endpointURL, endpointVersion string) error {
+	// check if storagecluster CRD is present, in case yes, we continue validation
+	// otherwise px was deployed using daemonset, we skip this validation
+	_, err := apiExtentions.GetCRD("storageclusters.core.libopenstorage.org", metav1.GetOptions{})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get storagecluster CRD. cause: %v", err)
+	} else if k8serrors.IsNotFound(err) {
+		return nil
+	}
+	pxOps, err := pxOperator.ListStorageClusters(schedops.PXNamespace)
+	if err != nil {
+		return err
+	}
+	if len(pxOps.Items) > 0 {
+		k8sVersion, err := k8sCore.GetVersion()
+		if err != nil {
+			return err
+		}
+		imageList, err := getImageList(endpointURL, endpointVersion, fmt.Sprintf("%s.%s.0", k8sVersion.Major, k8sVersion.Minor))
```

Comment:
Idk if this matters or not, but why do we hardcode 0 by default, k8s versions can consist of 1.2.3 and it might make a difference, if you hardcode 0 at the end like so 1.2.0, we might potentially have issues with some k8s version dependent components, I would ask Piyush about this.